app/Services/Api/ApiEntityService.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Services\Api;
+
+use App\Models\MiscModel;
+use App\Models\Campaign;
+
+class ApiEntityService
+{
+    /**
+     */
+    public function saveEntity(array $entity, MiscModel $class, Campaign $campaign)
```

Comment:
The campaign param can be moved out and added with the CampaignAware trait, and called only once outside of the loop. I also think that moving the class out into it's own function returning $this would make it clearer, rather than having a method with many parameters.